win32/build/confutils.js

```diff
@@ -3626,6 +3626,12 @@ function ADD_MAKEFILE_FRAGMENT(src_file)
 	}
 }
 
+function SETUP_ZLIB_LIB(target, path_to_check)
+{
+	return ((PHP_ZLIB == "no" || PHP_ZLIB_SHARED) && CHECK_LIB("zlib_a.lib;zlib.lib", target, path_to_check)) ||
+		(PHP_ZLIB == "yes" && !PHP_ZLIB_SHARED);
```

Comment:
This might need some explanation. zlib (like a couple of other extensions) is a bit special, because if PHP with zlib as static extension, php8.dll provides the functions as if they had been built as shared extension. Thus, there is no need to involve zlib.lib.

Reply:
This is not correct; see https://github.com/php/php-src/pull/16801#discussion_r1843652081.